src/elements/content-sidebar/MetadataSidebarRedesign.tsx

```diff
@@ -4,29 +4,197 @@
+            setFile(fetchedFile);
+            if (shouldFetchMetadata && fetchedFile) {
+                fetchMetadata(fetchedFile);
+            } else {
+                setStatus('success');
+            }
+        },
+        [fetchMetadata, file],
+    );
+
+    const fetchFileErrorCallback = React.useCallback(
+        (e: ElementsXhrError, code: string) => {
+            setFile(undefined);
+            onApiError(e, code, messages.sidebarMetadataEditingErrorContent);
+        },
+        [onApiError],
+    );
+
+    React.useEffect(() => {
+        setStatus('loading');
+        api.getFileAPI().getFile(fileId, fetchFileSuccessCallback, fetchFileErrorCallback, {
+            fields: [FIELD_IS_EXTERNALLY_OWNED, FIELD_PERMISSIONS],
+            refreshCache: true,
+        });
+        // this effect is suppose to run only once when component is loaded for a given file
+        // when all dependencies are listed components goes into infinite rerender loop
+        // even if functions are wrapped in useCallback
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [fileId]);
+
+    return {
+        status,
+        file,
+        errorMessage,
+        templates,
+    };
+}
+
+function MetadataSidebarRedesign({ api, fileId, onError, isFeatureEnabled }: MetadataSidebarRedesignProps) {
+    const [selectedTemplates, setSelectedTemplates] = React.useState<Array<MetadataTemplate>>([]);
+
+    const { templates, errorMessage, status } = useDataFetching(api, fileId, onError, isFeatureEnabled);
+
     return (
-        <div className="bcs-MetadataSidebarRedesign">
-            <h3>
-                <FormattedMessage {...messages.sidebarMetadataTitle} />
-            </h3>
-            <hr />
-            <p>Hello from Metadata Sidebar redesign</p>
+        <div className="bcs-MetadataSidebarRedesign bcs-content">
+            <div className="bcs-MetadataSidebarRedesign-header">
+                <Text as="h3" variant="titleLarge">
+                    <FormattedMessage {...messages.sidebarMetadataTitle} />
+                </Text>
+                {status === 'success' && templates && (
+                    <AddMetadataTemplateDropdown
+                        availableTemplates={templates}
+                        selectedTemplates={selectedTemplates}
+                        onSelect={(selectedTemplate): void => {
+                            setSelectedTemplates([...selectedTemplates, selectedTemplate]);
+                        }}
+                    />
+                )}
+            </div>
+            <div className="bcs-MetadataSidebarRedesign-content">
+                {status === 'error' && errorMessage && (
+                    <InlineError title={<FormattedMessage {...messages.error} />}>
+                        <FormattedMessage {...errorMessage} />
+                    </InlineError>
+                )}
+                {status === 'loading' && <LoadingIndicator />}
```

Comment:
if the status is in idle state first, does flickering occur here since loading indicator is only shown when it's loading?

Reply:
can't observe any flickering